File: front/src/app/models/CursoChat.ts
import { ClaseChat } from './ClaseChat';
import { MensajeChat } from './MensajeChat';

export class CursoChat {
	public idCurso: number;

	public clases: ClaseChat[];

	public mensajes: MensajeChat[];

	public nombreCurso: string;

	public fotoCurso: string;

	constructor(_idCurso: number, _clases: ClaseChat[], _mensajes: MensajeChat[], _nombreCurso: string, _fotoCurso: string) {
		this.idCurso = _idCurso;
		this.clases = _clases;
		this.mensajes = _mensajes;
		this.nombreCurso = _nombreCurso;
		this.fotoCurso = _fotoCurso;
	}
}

File: sov_school_back/back_chat/src/main/java/com/sovereingschool/back_chat/Models/CursoChat.java
package com.sovereingschool.back_chat.Models;

import java.io.Serializable;
import java.util.List;

import org.springframework.data.mongodb.core.mapping.Document;

import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "courses_chat")
public class CursoChat implements Serializable {
    @Id
    private String id;

    /** Curso ID */
    private Long idCurso;

    /** Lista de ClaseChat */
    private List<ClaseChat> clases;

    /** Lista de ids de documentos de mensajes */
    private List<String> mensajes;

    /** ID del documento del ultimo mensaje */
    private String ultimo;

}